PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/CameraScript/RootCamera.rbxmx

```diff
@@ -90,18 +103,88 @@ local SetCameraOnSpawn = true
 
 local hasGameLoaded = false
 
+local GestureArea = nil
+
+do
+	-- Check for changes in ViewportSize to decide if PortraitMode
+	local function onWorkspaceChanged(property)
+		if property == 'CurrentCamera' then
+			local newCamera = workspace.CurrentCamera
+			if newCamera then
+				local size = newCamera.ViewportSize
+				PortraitMode = size.X < size.Y
+				DefaultZoom = PortraitMode and 25 or 12.5
+				newCamera.Changed:Connect(function(property)
+					if property == 'ViewportSize' then
+						size = newCamera.ViewportSize
+						PortraitMode = size.X < size.Y
+						DefaultZoom = PortraitMode and 25 or 12.5
+					end
+				end)
+			end
+		end
+	end
+	if workspace.CurrentCamera then
+		workspace.Changed:Connect(onWorkspaceChanged)
+		onWorkspaceChanged('CurrentCamera')
+	end
+
+	-- Setup gesture area that camera uses while DynamicThumbstick is enabled
+	local function OnCharacterAdded(character)
+		if PlayerGui then
+			local ScreenGui = Instance.new("ScreenGui")
+			ScreenGui.Parent = PlayerGui
+			
+			GestureArea = Instance.new("Frame")
+			GestureArea.Parent = ScreenGui
+			GestureArea.BackgroundTransparency = 1.0
+			GestureArea.Visible = true
+			GestureArea.BackgroundColor3 = Color3.fromRGB(0, 0, 0)
+			GestureArea.Size = UDim2.new(1, 0, .6, 0)
+			GestureArea.Position = UDim2.new(0, 0, 0, 0)
+			if workspace.CurrentCamera then
+				local size = workspace.CurrentCamera.ViewportSize
+				PortraitMode = size.X < size.Y
```

Comment:
Yeah, Kyle's API changes will be that central location. This is a hack in case those aren't in by the time this is in.